### PR TITLE
feat(aws-ec2): add support for UDP to SecurityGroupRule

### PR DESCRIPTION
**Pull request by ChintanRaval**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts [hunk 1]
```diff
@@ -234,7 +234,92 @@ export class TcpAllPorts implements IPortRange {
...
+  public toRuleJSON(): any {
+    return {
+      ipProtocol: Protocol.Udp,
+      fromPort: this.port,
+      toPort: this.port
+    };
+  }
+
+  public toString() {
+    return `${this.port}`;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Can you add the word UDP to this description string?

LIke `UDP ${this.port}`?

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts [hunk 2]
```diff
@@ -234,7 +234,92 @@ export class TcpAllPorts implements IPortRange {
...
+  public toRuleJSON(): any {
+    return {
+      ipProtocol: Protocol.Udp,
+      fromPort: this.startPort,
+      toPort: this.endPort
+    };
+  }
+
+  public toString() {
+    return `${this.startPort}-${this.endPort}`;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Same here

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts [hunk 3]
```diff
@@ -234,7 +234,92 @@ export class TcpAllPorts implements IPortRange {
...
+  public toRuleJSON(): any {
+    return {
+      ipProtocol: Protocol.Udp,
+      fromPort: 0,
+      toPort: 65535
+    };
+  }
+
+  public toString() {
+    return 'ALL PORTS';
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Same here.

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts (hunk 1 above)
Comment (ChintanRaval) [reply packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: @rix0rrr Sure, while I prefix all the UDP-related `toString()` to use `UDP `, should I also do the same for the TCP ones? Just for consistency. We'd achieve uniqueness either way, but thought why should one be treated as a first-class citizen over the other?

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts [hunk 4]
```diff
@@ -234,7 +234,92 @@ export class TcpAllPorts implements IPortRange {
...
+  public toRuleJSON(): any {
+    return {
+      ipProtocol: Protocol.Udp,
+      fromPort: this.port,
+      toPort: this.port
+    };
+  }
+
+  public toString() {
+    return '{IndirectPort}';
```
Comment (ChintanRaval) [inline packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: @rix0rrr  Will this conflict with `TcpPortFromAttribute.toString()`'s output? If yes, then I'll prefix `UDP ` here as well else, I'll leave it as it is.

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts (hunk 2 above)
Comment (ChintanRaval) [reply packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Done.

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts (hunk 3 above)
Comment (ChintanRaval) [reply packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Done.

File: packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts (hunk 1 above)
Comment (ChintanRaval) [reply packages/@aws-cdk/aws-ec2/lib/security-group-rule.ts]: Also, is it worth it to do something like `${Protocol.Udp} ${this.port}` as compared to `UDP ${this.port}`?

**Comment by ChintanRaval** [comment]
@rix0rrr all done.

**Pull request merged**